File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int>numsSet;
        int count=1,largest=0;
        for(auto num:nums){
            numsSet.insert(num);
        }
        for(int i=0;i<nums.size();i++){
            if(numsSet.find(nums[i]-1)== numsSet.end()){
                int j=i;
                count=1;
                int sum=nums[j]+1;
                while(numsSet.find(sum)!= numsSet.end()){
                    count++;
                    sum++;
                }
                largest=max(largest,count);
            }
        }
        return largest;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        //store the elems in hashset
        //loop through hashset
        //check if the hashset contain elem-1. If not then we can assume it is the smallest elem and search for elems greater(elem+1) than that until it doesn't exist anymore and store the maxLength.
        HashSet<Integer> set=new HashSet<>();
        for(int i=0;i<nums.length;i++){
            set.add(nums[i]);
        }
        int max=1;
        if(nums.length==0) max=0;
        for(Integer it: set){
            if(!set.contains(it-1)){
                int i=it;
                while(set.contains(i+1)){
                    max=Math.max(max,++i-it+1);
                }
            }
        }
        return max;
    }
}